### PR TITLE
GEODE-5002: Create geode-test sub-module

### PR DESCRIPTION
**Pull request by sboorlagadda**
* created new gradle module to hold test framework
  * moved dunit test framework into geode-test
  * merged geode-junit and geode-old-versions into geode-test
  * modified each module's gradle build script to consider geode-test
  * changes necessary to support idea and eclipse IDEs

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-test/src/main/java/org/apache/geode/test/dunit/rules/DistributedTestRule.java [hunk 1]
```diff
@@ -212,7 +211,6 @@ public static void tearDownInVM() {
       DiskStoreObserver.setInstance(null);
       unregisterInstantiatorsInThisVM();
       DistributionMessageObserver.setInstance(null);
-      GlobalLockingDUnitTest.region_testBug32356 = null;
```
Comment (sboorlagadda) [inline geode-test/src/main/java/org/apache/geode/test/dunit/rules/DistributedTestRule.java]: This creates a cyclic dependency.

**Comment by metatype** [comment]
Reviewing now, but may take some time since there are a lot of changes.

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
-#Tue Jul 05 14:26:44 PDT 2016
+#Wed Apr 04 10:09:23 PDT 2018
```
Comment (PurelyApplied) [inline gradle/wrapper/gradle-wrapper.properties]: I don't think we're meant to update this file?  Changing the `distributionUrl` sounds dangerous.

Comment (sboorlagadda) [reply gradle/wrapper/gradle-wrapper.properties]: @PurelyApplied I will revert the changes to gradle.properties

**Comment by sboorlagadda** [comment]
@PurelyApplied & @kirklund Thanks for your review.

Precheckin is green. I would like to get this merged if I get one more 👍

**Comment by PurelyApplied** [comment]
+1!

**Pull request merged**